keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,969 @@
+
+### Non-Goals
+
+Our mission is to solve the most common use cases that cluster admins have.
+That is, we don't want to solve for every possible policy permutation a user
+can think of. Instead, we want to design an API that addresses 90-95% use cases
+while keeping the mental model easy to understand and use.
+
+Additionally, this proposal is squarely focused on solving the needs of the
+Cluster Administrator. It is not intended to solve:
+1. Logging / error reporting for Network Policy
+2. Kubernetes Node policies
+3. New policy selectors (services, service accounts, etc.)
+4. Tools/CLI to debug/explain NetworkPolicies, cluster-scoped policies and their
+   impact on workloads.
+
+## Proposal
+
+In order to achieve the two primary broad use cases for a cluster admin to
+secure K8s clusters, we propose to introduce the following two new resources
+under `networking.k8s.io` API group:
+- ClusterNetworkPolicy
+- DefaultNetworkPolicy
+
+### ClusterNetworkPolicy resource
+
+A ClusterNetworkPolicy resource will help the administrators set strict
+security rules for the cluster, i.e. a developer CANNOT override these rules
+by creating NetworkPolicies that applies to the same workloads as the
+ClusterNetworkPolicy does.
+
+Unlike the NetworkPolicy resource in which each rule represents a allowed
+traffic, ClusterNetworkPolicy will enable administrators to set `Allow` or
+`Deny` as the action of each rule. ClusterNetworkPolicy rules should be read
+as-is, i.e. there will not be any implicit isolation effects for the Pods
+selected by the ClusterNetworkPolicy, as opposed to what NetworkPolicy rules imply.
+
+In terms of precedence, the aggregated `Deny` rules (all ClusterNetworkPolicy
+rules with action `Deny` in the cluster combined) should be evaluated before
+aggregated ClusterNetworkPolicy `Allow` rules, followed by aggregated
+NetworkPolicy rules in all Namespaces. As such, the `Deny` rules have the
+highest precedence, which prevents them to be unexpectedly overwritten.
+
+ClusterNetworkPolicy `Deny` rules are useful for administrators to explicitly
+block traffic from malicious clients, or workloads that poses security risks.
+Those traffic restrictions can only be lifted once the`Deny` rules are deleted
+or modified. On the other hand, the `Allow` rules can be used to call out
+traffic in the cluster that needs to be allowed for certain components to work
+as expected (egress to CoreDNS for example). Those traffic should not be blocked
+when developers apply NetworkPolicy to their Namespaces which isolates the workloads.
+
+### DefaultNetworkPolicy resource
+
+A DefaultNetworkPolicy resource will help the administrators set baseline security
+rules for the cluster, i.e. a developer CAN override these rules by creating
+NetworkPolicies that applies to the same workloads as the DefaultNetworkPolicy does.
+
+DefaultNetworkPolicy works just like NetworkPolicy except that it is cluster-scoped.
```

Comment:
Yeah, I agree, although I would suggest ClusterDefaultNetworkPolicy (in case we ever find a need for a namespaced DefaultNetworkPolicy). It would match the standard naming convention of Foo and ClusterFoo used elsewhere.